xdatasets/spatial.py

```diff
@@ -1,110 +1,128 @@
 
-
-    indexer = shape_bbox_indexer(ds, space['geometry'])
+    indexer = shape_bbox_indexer(ds, space["geometry"])
     ds_copy = ds.isel(indexer).copy()
-    
+
     arrays = []
-    pbar = tqdm(space['geometry'].iterrows())
+    pbar = tqdm(space["geometry"].iterrows())
     for idx, row in pbar:
-        item = row[space['unique_id']] if space['unique_id'] != None and space['unique_id'] in row else idx
-        pbar.set_description(f"Spatial operations: processing polygon {item} with {dataset_name}")
-
-        geom = space['geometry'].iloc[[idx]]
+        item = (
+            row[space["unique_id"]]
+            if space["unique_id"] is not None and space["unique_id"] in row
+            else idx
+        )
+        pbar.set_description(
+            f"Spatial operations: processing polygon {item} with {dataset_name}"
+        )
+
+        geom = space["geometry"].iloc[[idx]]
         da = bbox_ds(ds_copy, geom)
 
         # Average data array over shape
-            #da = average_shape(da, shape=geom)
+        # da = average_shape(da, shape=geom)
         with HiddenPrints():
-            ds_weights  = create_weights_mask(da.isel(time=0), geom)
-        if space['averaging'] is True:   
+            ds_weights = create_weights_mask(da.isel(time=0), geom)
+        if space["averaging"] is True:
             da = aggregate(da, ds_weights)
         else:
             da = xr.merge([da, ds_weights])
             da = da.where(da.weights.notnull(), drop=True)
         da = da.expand_dims({"geom": geom.index.values})
         arrays.append(da)
 
-    data = xr.concat(arrays, dim='geom')
+    data = xr.concat(arrays, dim="geom")
 
-    if 'unique_id' in space:
+    if "unique_id" in space:
         try:
             data = data.swap_dims({"geom": space["unique_id"]})
-            data = data.drop('geom')
-
-            if space['unique_id'] not in data.coords:
-                data = data.assign_coords({space['unique_id']: (space['unique_id'],
-                                                        space['geometry'][space['unique_id']])})   
-        except:
+            data = data.drop_vars("geom")
+
+            if space["unique_id"] not in data.coords:
+                data = data.assign_coords(
+                    {
+                        space["unique_id"]: (
+                            space["unique_id"],
+                            space["geometry"][space["unique_id"]],
+                        )
+                    }
+                )
+        except KeyError:
             pass
     return data
 
 
 def clip_by_point(ds, space, dataset_name):
-
-    #TODO : adapt logic for coordinate names
+    # TODO : adapt logic for coordinate names
 
     logger = logging.getLogger()
-    logging.info(f"Spatial operations: processing points with {dataset_name}")
+    logger.info(f"Spatial operations: processing points with {dataset_name}")
```

Comment:
`logger` is never set elsewhere, for example in `clip_by_bbox`. Should we instead git it as it was, but remove `logger`?

Reply:
If we aren't going to employ conventional logging practices (setting loggers by file/scope), we're probably safer not instantiating `logger` at all here.